java/ql/src/experimental/semmle/code/java/Logging.qll

```diff
@@ -0,0 +1,26 @@
+/**
+ * Provides classes and predicates for working with logger's and their components.
+ */
+
+import java
+
+/** Models a call to a logging method.. */
+class LoggingCall extends MethodAccess {
+  LoggingCall() {
+    exists(RefType t, Method m |
+      t.hasQualifiedName("org.apache.logging.log4j", ["Logger", "Category", "LogBuilder"]) or
+      t.hasQualifiedName("org.apache.commons.logging", "Log") or
+      t.hasQualifiedName("org.jboss.logging.Logger", "BasicLogger") or
+      t.hasQualifiedName("org.slf4j.spi", "LoggingEventBuilder") or
+      t.hasQualifiedName("org.slf4j", "Logger") or
+      t.hasQualifiedName("org.scijava.log", "Logger")
+    |
+      m.getDeclaringType().(RefType).extendsOrImplements(t) and
+      m.getReturnType() instanceof VoidType and
```

Comment:
This is too broad. We should specify the method names, which will presumably be different for each framework.

Reply:
@owen-mc Can you provide me a FP due to this. All of the methods return a void type regardless of their name. Those which should not be sinks either return something else which is non-void or the argument accepted by those methods is something the taint won't flow into unless you explicitly mark it as such.

Reply:
@porcupineyhairs My objection isn't that doing it this way causes FPs. My main objection is that it's hard for the reader to know what methods are included. It also makes it harder to add to this list in future. If you disagree you are free to ignore this suggestion.

Reply:
okay now I get your point. I would list the name explicitly then.

Reply:
@owen-mc The list of all method names exceeds more than 20 items. I think keeping this block as is better than adding 20 entries. A comment line explaining the block would serve the purpose of listing all method names independently here instead. WDYT?